### PR TITLE
Private events - Rails Associations

### PR DESCRIPTION
**Pull request by mohammadumar28**
- [x] Setup project environment
- [x] Install Styleint and Rubocop
- [x] Create Models for User, Events and Stats
- [x] Create Associations between the models
- [x] Implement Sign up and Login feature
- [x] Style the views
- [x] Resolve Rubocop offences

The project requirements have been completed, hence request for the project review.

### CODE REVIEW

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    <h4 class="text-center">Past events</h4>
+    <% @user.stats.each do |att| %>
+      <% if att.attended_event.date < Time.now %>
+        <div class="card mt-2">
+          <div class="card-body">
+            <h5 class="card-title"><%= att.attended_event.title %></h5>
+            <p class="card-text"><%= att.attended_event.description %></p>
+            <div class="alert alert-info" role="alert">
+              <%= att.attended_event.date.strftime("%B %d, %Y") %>
+            </div>
```
Comment (kubilaycaglayan) [inline app/views/users/show.html.erb:16]: :bulb: There shouldn't be any logic in the views 🙏

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      <% end %>
+    <% end %>
+  </div>
+
+  <div class="col-6">
+    <h4 class="text-center">Upcoming Events</h4>
+    <% @user.stats.each do |att| %>
+      <% if att.attended_event.date > Time.now %>
+        <div class="card mt-2">
+          <div class="card-body">
```
Comment (kubilaycaglayan) [inline app/views/users/show.html.erb:31]: :bulb: There shouldn't be any logic in the views :pray:

File: app/controllers/events_controller.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+  def new
+    @user = current_user
+    @event = @user.events.new
+  end
+
+  def create
+    @user = current_user
+    @event = @user.events.new(event_params)
+    redirect_to @event if @event.save
```
Comment (kubilaycaglayan) [inline app/controllers/events_controller.rb:15]: :bulb: Uses build method.

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
+class UsersController < ApplicationController
+  before_action :set_user, only: [:show]
+
+  def index
+    @users = User.all
+  end
+
+  def show; end
```
Comment (kubilaycaglayan) [inline app/controllers/users_controller.rb:8]: :bulb: Let's spice up here according to the requirements, please :pray:

**Review by kubilaycaglayan — CHANGES_REQUESTED** [review]
Hello Team 👋
Status: Changes Required ♻️

Great job so far! Your page looks really good. Thanks for spending extra time to add styling. It is easier to navigate through the website... 👍👏

Let's cover some cases together ⏬
All of the checkboxes below are [REQUIRED] except one optional case.

### Controllers
#### events#create

- [ ] uses #build method on association, i.e. `current_user.events.build(event_params)` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/build_method_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

#### users#show
- [x] lists events created by user, i.e. `@created_events =  current_user.events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/current_user_show_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
- [x] lists events previoulsy attendented by user **using event model scope**, i.e. `@past_events = current_user.attended_events.previous` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
- [x] lists upcoming events previously attended by user **using event model scope**, i.e. `@upcoming_events = current_user.attended_events.upcoming` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

### No Logic In The Views
💡 Please clean all the logic from the views until none left.

### Database Foreign Keys

- [ ] the events_attendation/events_users/(any other name for join table between users and events) table has column `event_id` of type `integer`
- [ ] the events_attendation/events_users/(any other name for join table between users and events) table has column `user_id` of type `integer`
 💡 Currently you have `attended_event_id`, and `attendee_id` columns. Please change their names accordingly. This change is required for naming conventions. 😇

### Tests
- [ ] models have unit tests for validations and associations (Rspec) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/unit_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
💡 Please add validation tests. Currently, you have only 8 test cases and all of them are association tests.

- [ ] authentication have has integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
💡 Please add authentication tests.

- [ ] event managment features have integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
💡 Please add tests for event creation.

- [ ] [OPTIONAL] You can add invite feature 😊🎉

#### That's it, team! After you make the changes, you can submit for another code review. ♻️

With all my respect. 🙏

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team    
#### Great Job so far, everything works fine, but there are a few more things required. 
### General 
- [x]  I should be able to attend events created by other users. 
### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Review by andresporras3423 — CHANGES_REQUESTED** [review]
### Status: Changes Required :recycle:
**Hello Aldana Longhi & Umar Chingani** 
Great job with your new submission :muscle:  :muscle: 
Your page meets most of the requirements :clap: :clap: :clap:  
Just a couple of details to fix:

- [ ] test retrieve of past and upcoming events in your unit tests
- [ ] Your project use yarn so the next command was necessary for me to run your project: 'yarn install --check-files'. In the README file, inside instructions to run locally, add the command.

Please request another review once you are ready! :+1: 

Happy coding :relaxed:

Oscar Andrés Russi Porras

**Review by MuriloRoque — APPROVED** [review]
# Status: Approved :green_circle: 

### Hello @mohammadumar28 and @AlduLonghi :wave: 

### Thank you for making all of the requested changes. Your project is approved :clap: :clap: :clap: 

### You may merge, good luck with your next projects :rocket: 

Murilo Roque Paiva da Silva

- Github: [@muriloroque](https://github.com/MuriloRoque)
- Linkedin: [MuriloRoque](https://www.linkedin.com/in/murilo-roque-b1268741/)

[Twitter Follow](https://twitter.com/MuriloRoquePai1)

**Pull request merged**